dev-docs/RFCs/proposals/color-filter-extension-rfc.md

```diff
@@ -0,0 +1,126 @@
+
+#### Implementation
+
+- Introduce a `ColorFilterExtension` that accepts a ShaderPass as an input parameter.
+- Modify the layer's fragment shader by integrating the provided ShaderPass at render time.
+
+```js
+import {
+  brightnessContrast,
+  hueSaturation,
+} from '@luma.gl/shadertools'
+
+new Tile3DLayer({
+    extensions: [new ColorFilterExtension(brightnessContrast)]
+    brightnessContrast: {
+        brightness: 0.5,
+        contrast: 0.5
+    }
+})
+```
+
+Multiple filters can be applied with more than one instance:
+
+```js
+{
+    extensions: [new ColorFilterExtension(hueSaturation), new ColorFilterExtension(brightnessContrast)]
+    hueSaturation: ...,
+    brightnessContrast: ...,
+}
+```
+
+#### Pros and Cons
+
+**Pros:**
+- Flexibility: Enables users to create custom effects.
+- Compatibility: Integrates seamlessly with existing `luma.gl` modules.
+
+**Cons:**
+- Usage Complexity: Requires users to have a good understanding of `luma.gl` shaders.
+- Typings Complexity: Determining extension properties is complex since the ShaderPass dictates them.
+
+### Design 2: Built-in Shaders in Extension
+
+#### Overview
+
+This design incorporates a set of predefined shaders within the extension, addressing common filters like tint, brightness, and saturation.
+
+#### Implementation
+
+- Equip the `ColorFilterExtension` with built-in shader modules.
+- Allow users to select a filter and modify its parameters (e.g., intensity).
+
+```js
+new Tile3DLayer({
+    extensions: [new ColorFilterExtension()]
+    brightnessContrast: {
+        brightness: 0.5,
+        contrast: 0.5
+    }
+})
+```
+
+All single pass "filter" shaders would already be built-in: brightnessContrast, hueSaturation, vibrance, tint (mix original colors), fill (set colors to single hue), sepia, colorHalftone, and dotScreen. So, multiple filters can be applied with one instance:
+
+```js
+new Tile3DLayer({
+    extensions: [new ColorFilterExtension()]
+    hueSaturation: ...,
+    brightnessContrast: ...,
+})
+```
+
+##### Props
+
+- shaders (`Array<String>`):
+
+Should a user need to customize the shader order we could have an `shaders` extension prop to specify which shaders to load and in which order. 
+
+Usage: `new ColorFilterExtension({shaders: ['hueSaturation', 'tint']})`
+
+Shader order can give a different result, so this may be useful in the event a user wanted to first make their layer greyscale and then tint it as opposed to tinting before desaturating.
+
+The same prop may be useful for cherry picking which shaders to apply when users are concerned with shader overhead.
+
+#### Pros and Cons
+
+**Pros:**
+- Ease of Use: Simplifies the process for users unfamiliar with shader programming.
+- Consistency: Ensures standardized performance and appearance.
+
+**Cons:**
+- Limited Flexibility: Confines users to predefined filters.
```

Comment:
For me this is too limiting. I don't find importing a shader module from luma and passing it to the `ColorFilterExtension` constructor too onerous. It is also what we already have in the `PostProcessEffect`